packages/sitecore-jss-cli/src/process-env.ts

```diff
@@ -0,0 +1,26 @@
+import path from 'path';
+import * as dotenv from 'dotenv';
+import dotenvExpand from 'dotenv-expand';
+
+/**
+ * @param {string} dir The directory containing the .env files to load.
+ */
+export default function processEnv(dir: string) {
+  // replicate Next.js handling/behavior
+  // https://github.com/vercel/next.js/blob/v10.0.5/packages/next-env/index.ts#L80-L90
+  const mode = process.env.NODE_ENV;
+  const dotenvFiles = [
+    `.env.${mode}.local`,
+    // Don't include `.env.local` for `test` environment
+    // since normally you expect tests to produce the same
+    // results for everyone
+    mode !== 'test' && '.env.local',
+    `.env.${mode}`,
+    '.env',
+  ].filter(Boolean) as string[];
```

Comment:
I know this code snippet is from the nextjs repo, but I'm wondering to myself why they took the extra step of casting to a string[]. Running the original string array through `Boolean` should just remove empty strings from the array, not change the type. Do you see what edge cases this is covering?

Reply:
I assumed it was for this:

`mode !== 'test' && '.env.local'`

I'm not a typescript expert but I assume that the `Boolean` filter would remove the `false` when `mode == 'test'`, and the cast is needed because otherwise it's a `string | Boolean` array?